### PR TITLE
Support blobby image types

### PR DESCRIPTION
**Pull request by achilleas-k**
## Blobby image types

Added the new image types supported in composer to the list of allowed image type names:
- edge-container
- guest-image (qcow2)
- image-installer
- vsphere (vmdk)

Reordered and changed the enum in the api spec yaml to make it easier to see differences moving forward (one item per line) and to add comments to indicate backwards-compatible aliases.

## Quota file handling

Extended error checking and messages for the quota file.

- Return error if the file can't be read for more specific error reporting
- Return error message prefix indicating where the failure came from for easier troubleshooting

### CODE REVIEW

File: internal/v1/api.yaml [hunk 1]
```diff
@@ -378,7 +378,21 @@ components:
...
+        - aws
+        - azure
+        - edge-commit
+        - edge-container
+        - edge-installer
+        - gcp
+        - guest-image
+        - image-installer
+        - vsphere
+        # backwards compatible aliases
```
Comment (Gundersanne) [inline internal/v1/api.yaml:391]: good comment :)

File: internal/v1/api.go [hunk 2]
```diff
@@ -182,11 +182,15 @@ const (
 	ImageTypes_aws                 ImageTypes = "aws"
 	ImageTypes_azure               ImageTypes = "azure"
 	ImageTypes_edge_commit         ImageTypes = "edge-commit"
+	ImageTypes_edge_container      ImageTypes = "edge-container"
```
Comment (Gundersanne) [inline internal/v1/api.go:185]: so this was disabled anyway the whole time :|

**Review by Gundersanne — DISMISSED** [review]
thank you!

File: internal/v1/api.go (hunk 2 above)
Comment (achilleas-k) [reply internal/v1/api.go:185]: looks like it.  which reminds me, it also needs to be added to the upload target switch.

**Comment by achilleas-k** [comment]
Pushed a small (squashed) change here: https://github.com/osbuild/image-builder/pull/309/files#diff-d0d6ee194b461efad1c4eea103b4dbfb740227c79b5d93daa697144bfb7d1c9cR527-R528

**Review by Gundersanne — APPROVED** [review]

**Comment by achilleas-k** [comment]
What's up with the google cloud test?  Is it something I did?

**Comment by Gundersanne** [comment]
> What's up with the google cloud test? Is it something I did?

nah, it got fixed in #307

**Comment by Gundersanne** [comment]
The admin merge checkbox doesn't work :|

**Comment by achilleas-k** [comment]
Rebased for fix

**Review by Gundersanne — APPROVED** [review]

**Comment by achilleas-k** [comment]
We are green :)

**Pull request merged**